src/rendering/renderers/pool/PoolLockCountdownRenderer.ts

```diff
@@ -31,10 +71,28 @@ export class PoolLockCountdownRenderer extends BaseRenderer {
 
     protected update(
         context: RenderingContext,
-        container: Container,
+        container: Container
     ): Container {
         if (!context.pool) return this.hideContainer(container)
 
+        const { lockDate } = context.pool
+        if (DateUtils.nowUnixTS() > lockDate) return this.hideContainer(container)
+
+        this.updateBackground(context, container)
+        this.updateText(context, container)
+
+        this._lockDate = context.pool.lockDate
+        this.visitPeriod(() => this.updateCountdown())
```

Comment:
let's start single period visitor in constructor
and just save visitor function on each render
so period visitor will call latest visit function